src/Compilers/VisualBasic/Test/Semantic/Semantics/InterpolatedStringTests.vb

```diff
@@ -1140,6 +1140,151 @@ BC30491: Expression does not produce a value.
         End Sub
 
         <Fact>
+        Public Sub FlowAnalysis_Warning_InterpoledVariableUsedBeforeBeingAssigned()
+
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntimeAndReferences(
+<compilation>
+    <%= _formattableStringSource %>
+    <file name="a.vb">
+Imports System
+Imports System.Console
+
+Module Program
+    Sub Main()
+        Dim v As Object
+
+        Write($"{v}")
+    End Sub
+End Module
+    </file>
+</compilation>)
+
+            AssertTheseCompileDiagnostics(compilation,
+<expected>
+BC42104: Variable 'v' is used before it has been assigned a value. A null reference exception could result at runtime.
+        Write($"{v}")
+                 ~
+</expected>)
+
+        End Sub
+
+        <Fact>
+        Public Sub FlowAnalysis_InterpolatedLocalConstNotConsideredUnused()
+
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntimeAndReferences(
+<compilation>
+    <%= _formattableStringSource %>
+    <file name="a.vb">
+Imports System
+Imports System.Console
+
+Module Program
+    Sub Main()
+        Const v As Object = Nothing
+
+        Write($"{v}")
+    End Sub
+End Module
+    </file>
+</compilation>)
+
+            AssertNoDiagnostics(compilation)
+
+        End Sub
+
+        <Fact>
+        Public Sub FlowAnalysis_AnalyzeDataFlowReportsCorrectResultsForVariablesUsedInInterpolations()
+
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntimeAndReferences(
+<compilation>
+    <%= _formattableStringSource %>
+    <file name="a.vb">
+Imports System
+Imports System.Console
+
+Module Program
+    Sub Main()
+        Dim v As Object = Nothing
+
+        Write($"Expression {v} is not a value.")
+
+        WriteLine(v)
+    End Sub
+End Module
+    </file>
+</compilation>)
+
+            AssertNoDiagnostics(compilation)
+
+            Dim mainTree = Aggregate t In compilation.SyntaxTrees Where t.FilePath = "a.vb" Into [Single]()
+            Dim root = mainTree.GetRoot()
+            Dim sm = compilation.GetSemanticModel(mainTree)
+
+            Dim vSymbol = CType(sm.GetDeclaredSymbol(root.DescendantNodes().OfType(Of ModifiedIdentifierSyntax).Single()), ILocalSymbol)
+
+            Dim analysis = sm.AnalyzeDataFlow(root.DescendantNodes().OfType(Of ExpressionStatementSyntax).First())
```

Comment:
Added an assert in this test and the one below it on the text of the node being analyzed.